app/models/patient.rb

```diff
@@ -368,6 +371,47 @@ def latest_transfer
     end
   }
 
+  # Gets the current date in the patient's timezone
+  def curr_date_in_timezone
+    Time.now.getlocal(address_timezone_offset)
+  end
+
+  # Checks is at the end of or past their monitoring period
+  def end_of_monitoring_period?
+    return false if continuous_exposure
+
+    monitoring_period_days = ADMIN_OPTIONS['monitoring_period_days'].days
+    monitoring_end_date = nil
+
+    # If there is a last date of exposure - base monitoring period off of that date
+    if !last_date_of_exposure.nil?
+      monitoring_end_date = last_date_of_exposure.beginning_of_day + monitoring_period_days
+    elsif !created_at.nil?
```

Comment:
When is it possible for a patient in the database have a `created_at` value of `nil`?

Reply:
Shouldn't be, just there as a safety measure really. We do a similar check in the `end_of_monitoring` instance method in this file.

Reply:
Yeah I would suggest that both of them be changed because that check just returns `true` It would simplify this method a bit also.

```ruby
  def end_of_monitoring_period?
    return false if continuous_exposure

    monitoring_period_days = ADMIN_OPTIONS['monitoring_period_days'].days

    # If there is a last date of exposure - base monitoring period off of that date
    if !last_date_of_exposure.nil?
      monitoring_end_date = last_date_of_exposure.beginning_of_day + monitoring_period_days
    else
      # Otherwise, if there is no last date of exposure - base monitoring period off of creation date
      monitoring_end_date = created_at.beginning_of_day + monitoring_period_days
    end

    # If it is the last day of or past the monitoring period
    # NOTE: beginning_of_day is used as monitoring period is based of date not the specific time
    curr_date_in_timezone.beginning_of_day >= monitoring_end_date
  end
```

Reply:
Updated this method as well as `end_of_monitoring` to no longer include unnecessary `created_at` checks.